### ISSUE TITLE
Mono: Editor crashes sometimes on the constructor of an object

### ISSUE THREAD

**Issue by MightyPrinny** [godotengine/godot#38095]
**Godot version:**
3.2.2 beta1 

**OS/device including version:**
MXLinux 19 64 bits

**Issue description:**
The editor crashes after I change some code, rebuild the project and open a scene at the constructor of some classes.

I'll try to make an MRP as soon as I can but maybe the stack trace has a clue.

It can happen on many of my classes, I have fields of types that don't inherit from Godot.Object initialized in their declaration, but this didn't cause editor crashes so consistently before.

Here's what the console prints when the crash happens

```
=================================================================
	Native Crash Reporting
=================================================================
Got a SIGSEGV while executing native code. This usually indicates
a fatal error in the mono runtime or one of the native libraries 
used by your application.
=================================================================

=================================================================
	Native stacktrace:
=================================================================
	0xd10b84 - ./Godot : (null)
	0xd10efc - ./Godot : (null)
	0xd039a8 - ./Godot : (null)
	0xc7c211 - ./Godot : (null)
	0x30175a0 - ./Godot : _ZN11GDMonoClass13get_mono_typeEv

=================================================================
	Telemetry Dumper:
=================================================================
Pkilling 0x7f5804657700 from 0x7f58370fc880
Pkilling 0x7f5828ffe700 from 0x7f58370fc880
Pkilling 0x7f57eae9e700 from 0x7f58370fc880
Pkilling 0x7f5804361700 from 0x7f58370fc880
Pkilling 0x7f57ea12c700 from 0x7f58370fc880
Pkilling 0x7f57eb4a1700 from 0x7f58370fc880
Pkilling 0x7f57eb09f700 from 0x7f58370fc880
Pkilling 0x7f57eac9d700 from 0x7f58370fc880
* Assertion: should not be reached at threads.c:6258
Entering thread summarizer pause from 0x7f58370fc880
Finished thread summarizer pause from 0x7f58370fc880.

Waiting for dumping threads to resume

=================================================================
	External Debugger Dump:
=================================================================
[New LWP 10261]
[New LWP 10266]
[New LWP 10267]
[New LWP 10268]
[New LWP 10269]
[New LWP 10270]
[New LWP 10271]
[New LWP 10272]
[New LWP 10273]
[New LWP 10274]
[New LWP 10278]
[New LWP 10452]
[New LWP 10454]
[New LWP 10498]
[New LWP 10499]
[New LWP 10500]
[New LWP 10512]
[New LWP 10513]
[Thread debugging using libthread_db enabled]
Using host libthread_db library "/lib/x86_64-linux-gnu/libthread_db.so.1".
0x00007f583968f0ca in waitpid () from /lib/x86_64-linux-gnu/libpthread.so.0
  Id   Target Id                                           Frame 
* 1    Thread 0x7f58370fc880 (LWP 10256) "Godot"           0x00007f583968f0ca in waitpid () from /lib/x86_64-linux-gnu/libpthread.so.0
  2    Thread 0x7f5832a46700 (LWP 10261) "Godot"           0x00007f583968d896 in do_futex_wait.constprop () from /lib/x86_64-linux-gnu/libpthread.so.0
  3    Thread 0x7f583119f700 (LWP 10266) "Godot:disk$0"    0x00007f583968b00c in pthread_cond_wait@@GLIBC_2.3.2 () from /lib/x86_64-linux-gnu/libpthread.so.0
  4    Thread 0x7f5830599700 (LWP 10267) "Godot"           0x00007f583968d896 in do_futex_wait.constprop () from /lib/x86_64-linux-gnu/libpthread.so.0
  5    Thread 0x7f5830157700 (LWP 10268) "Godot"           0x00007f583968ebf0 in nanosleep () from /lib/x86_64-linux-gnu/libpthread.so.0
  6    Thread 0x7f5830116700 (LWP 10269) "Godot"           0x00007f58393f0720 in nanosleep () from /lib/x86_64-linux-gnu/libc.so.6
  7    Thread 0x7f582a7ff700 (LWP 10270) "SGen worker"     0x00007f583968b00c in pthread_cond_wait@@GLIBC_2.3.2 () from /lib/x86_64-linux-gnu/libpthread.so.0
  8    Thread 0x7f5828ffe700 (LWP 10271) "Finalizer"       0x00007f583968d896 in do_futex_wait.constprop () from /lib/x86_64-linux-gnu/libpthread.so.0
  9    Thread 0x7f5807d7e700 (LWP 10272) "Godot"           0x00007f583968ebf0 in nanosleep () from /lib/x86_64-linux-gnu/libpthread.so.0
  10   Thread 0x7f5807d37700 (LWP 10273) "Godot"           0x00007f583968ebf0 in nanosleep () from /lib/x86_64-linux-gnu/libpthread.so.0
  11   Thread 0x7f5807531700 (LWP 10274) "Godot"           0x00007f583968ebf0 in nanosleep () from /lib/x86_64-linux-gnu/libpthread.so.0
  12   Thread 0x7f5806c56700 (LWP 10278) "Godot"           0x00007f583968d896 in do_futex_wait.constprop () from /lib/x86_64-linux-gnu/libpthread.so.0
  13   Thread 0x7f57eb09f700 (LWP 10452) "Thread Pool Wor" 0x00007f583968da82 in do_futex_wait () from /lib/x86_64-linux-gnu/libpthread.so.0
  14   Thread 0x7f57eae9e700 (LWP 10454) "Thread Pool Wor" 0x00007f583968da82 in do_futex_wait () from /lib/x86_64-linux-gnu/libpthread.so.0
  15   Thread 0x7f57eb4a1700 (LWP 10498) "Godot"           0x00007f583968b3f9 in pthread_cond_timedwait@@GLIBC_2.3.2 () from /lib/x86_64-linux-gnu/libpthread.so.0
  16   Thread 0x7f57ea12c700 (LWP 10499) "Thread Pool Wor" 0x00007f583968da82 in do_futex_wait () from /lib/x86_64-linux-gnu/libpthread.so.0
  17   Thread 0x7f5804361700 (LWP 10500) "Thread Pool Wor" 0x00007f583968da82 in do_futex_wait () from /lib/x86_64-linux-gnu/libpthread.so.0
  18   Thread 0x7f5804657700 (LWP 10512) "Thread Pool Wor" 0x00007f583968da82 in do_futex_wait () from /lib/x86_64-linux-gnu/libpthread.so.0
  19   Thread 0x7f57eac9d700 (LWP 10513) "Thread Pool Wor" 0x00007f583968da82 in do_futex_wait () from /lib/x86_64-linux-gnu/libpthread.so.0

Thread 19 (Thread 0x7f57eac9d700 (LWP 10513)):
#0  0x00007f583968da82 in do_futex_wait () from /lib/x86_64-linux-gnu/libpthread.so.0
#1  0x00007f583968db93 in __new_sem_wait_slow () from /lib/x86_64-linux-gnu/libpthread.so.0
#2  0x0000000000e91154 in mono_os_sem_timedwait (flags=MONO_SEM_FLAGS_ALERTABLE, timeout_ms=24033, sem=0x4bf4e88 <worker+72>) at ../../mono/utils/mono-os-semaphore.h:252
#3  mono_coop_sem_timedwait (sem=0x4bf4e88 <worker+72>, flags=MONO_SEM_FLAGS_ALERTABLE, timeout_ms=<optimized out>) at ../../mono/utils/mono-coop-semaphore.h:55
#4  worker_park () at threadpool-worker-default.c:391
#5  worker_thread (unused=<optimized out>) at threadpool-worker-default.c:490
#6  worker_thread (unused=unused@entry=0x0) at threadpool-worker-default.c:457
#7  0x0000000000ea1a55 in start_wrapper_internal (stack_ptr=0x7f57eac9e000, start_info=0x0) at threads.c:1226
#8  start_wrapper (data=0x7f57d8037f10) at threads.c:1299
#9  0x00007f5839684fa3 in start_thread () from /lib/x86_64-linux-gnu/libpthread.so.0
#10 0x00007f58394234cf in clone () from /lib/x86_64-linux-gnu/libc.so.6

Thread 18 (Thread 0x7f5804657700 (LWP 10512)):
#0  0x00007f583968da82 in do_futex_wait () from /lib/x86_64-linux-gnu/libpthread.so.0
#1  0x00007f583968db93 in __new_sem_wait_slow () from /lib/x86_64-linux-gnu/libpthread.so.0
#2  0x0000000000e91154 in mono_os_sem_timedwait (flags=MONO_SEM_FLAGS_ALERTABLE, timeout_ms=18979, sem=0x4bf4e88 <worker+72>) at ../../mono/utils/mono-os-semaphore.h:252
#3  mono_coop_sem_timedwait (sem=0x4bf4e88 <worker+72>, flags=MONO_SEM_FLAGS_ALERTABLE, timeout_ms=<optimized out>) at ../../mono/utils/mono-coop-semaphore.h:55
#4  worker_park () at threadpool-worker-default.c:391
#5  worker_thread (unused=<optimized out>) at threadpool-worker-default.c:490
#6  worker_thread (unused=unused@entry=0x0) at threadpool-worker-default.c:457
#7  0x0000000000ea1a55 in start_wrapper_internal (stack_ptr=0x7f5804658000, start_info=0x0) at threads.c:1226
#8  start_wrapper (data=0x7f57d4001660) at threads.c:1299
#9  0x00007f5839684fa3 in start_thread () from /lib/x86_64-linux-gnu/libpthread.so.0
#10 0x00007f58394234cf in clone () from /lib/x86_64-linux-gnu/libc.so.6

Thread 17 (Thread 0x7f5804361700 (LWP 10500)):
#0  0x00007f583968da82 in do_futex_wait () from /lib/x86_64-linux-gnu/libpthread.so.0
#1  0x00007f583968db93 in __new_sem_wait_slow () from /lib/x86_64-linux-gnu/libpthread.so.0
#2  0x0000000000e91154 in mono_os_sem_timedwait (flags=MONO_SEM_FLAGS_ALERTABLE, timeout_ms=48945, sem=0x4bf4e88 <worker+72>) at ../../mono/utils/mono-os-semaphore.h:252
#3  mono_coop_sem_timedwait (sem=0x4bf4e88 <worker+72>, flags=MONO_SEM_FLAGS_ALERTABLE, timeout_ms=<optimized out>) at ../../mono/utils/mono-coop-semaphore.h:55
#4  worker_park () at threadpool-worker-default.c:391
#5  worker_thread (unused=<optimized out>) at threadpool-worker-default.c:490
#6  worker_thread (unused=unused@entry=0x0) at threadpool-worker-default.c:457
#7  0x0000000000ea1a55 in start_wrapper_internal (stack_ptr=0x7f5804362000, start_info=0x0) at threads.c:1226
#8  start_wrapper (data=0x7f57c8001cd0) at threads.c:1299
#9  0x00007f5839684fa3 in start_thread () from /lib/x86_64-linux-gnu/libpthread.so.0
#10 0x00007f58394234cf in clone () from /lib/x86_64-linux-gnu/libc.so.6

Thread 16 (Thread 0x7f57ea12c700 (LWP 10499)):
#0  0x00007f583968da82 in do_futex_wait () from /lib/x86_64-linux-gnu/libpthread.so.0
#1  0x00007f583968db93 in __new_sem_wait_slow () from /lib/x86_64-linux-gnu/libpthread.so.0
#2  0x0000000000e91154 in mono_os_sem_timedwait (flags=MONO_SEM_FLAGS_ALERTABLE, timeout_ms=27553, sem=0x4bf4e88 <worker+72>) at ../../mono/utils/mono-os-semaphore.h:252
#3  mono_coop_sem_timedwait (sem=0x4bf4e88 <worker+72>, flags=MONO_SEM_FLAGS_ALERTABLE, timeout_ms=<optimized out>) at ../../mono/utils/mono-coop-semaphore.h:55
#4  worker_park () at threadpool-worker-default.c:391
#5  worker_thread (unused=<optimized out>) at threadpool-worker-default.c:490
#6  worker_thread (unused=unused@entry=0x0) at threadpool-worker-default.c:457
#7  0x0000000000ea1a55 in start_wrapper_internal (stack_ptr=0x7f57ea12d000, start_info=0x0) at threads.c:1226
#8  start_wrapper (data=0x7f57d4001570) at threads.c:1299
#9  0x00007f5839684fa3 in start_thread () from /lib/x86_64-linux-gnu/libpthread.so.0
#10 0x00007f58394234cf in clone () from /lib/x86_64-linux-gnu/libc.so.6

Thread 15 (Thread 0x7f57eb4a1700 (LWP 10498)):
#0  0x00007f583968b3f9 in pthread_cond_timedwait@@GLIBC_2.3.2 () from /lib/x86_64-linux-gnu/libpthread.so.0
#1  0x0000000000f1df9b in mono_os_cond_timedwait (cond=cond@entry=0x4c00940 <sleep_cond>, mutex=mutex@entry=0x4c00980 <sleep_mutex>, timeout_ms=timeout_ms@entry=500) at mono-os-mutex.c:75
#2  0x0000000000f2a451 in mono_coop_cond_timedwait (cond=0x4c00940 <sleep_cond>, mutex=0x4c00980 <sleep_mutex>, timeout_ms=500) at ../../mono/utils/mono-coop-mutex.h:100
#3  sleep_interruptable (alerted=0x7f57eb4a0c7c, ms=500) at mono-threads.c:1641
#4  mono_thread_info_sleep (ms=ms@entry=500, alerted=alerted@entry=0x7f57eb4a0c7c) at mono-threads.c:1673
#5  0x0000000000e90a03 in monitor_thread (unused=0x1f4) at threadpool-worker-default.c:708
#6  monitor_thread (unused=unused@entry=0x0) at threadpool-worker-default.c:670
#7  0x0000000000ea1a55 in start_wrapper_internal (stack_ptr=0x7f57eb4a2000, start_info=0x0) at threads.c:1226
#8  start_wrapper (data=0xeda8490) at threads.c:1299
#9  0x00007f5839684fa3 in start_thread () from /lib/x86_64-linux-gnu/libpthread.so.0
#10 0x00007f58394234cf in clone () from /lib/x86_64-linux-gnu/libc.so.6

Thread 14 (Thread 0x7f57eae9e700 (LWP 10454)):
#0  0x00007f583968da82 in do_futex_wait () from /lib/x86_64-linux-gnu/libpthread.so.0
#1  0x00007f583968db93 in __new_sem_wait_slow () from /lib/x86_64-linux-gnu/libpthread.so.0
#2  0x0000000000e91154 in mono_os_sem_timedwait (flags=MONO_SEM_FLAGS_ALERTABLE, timeout_ms=30364, sem=0x4bf4e88 <worker+72>) at ../../mono/utils/mono-os-semaphore.h:252
#3  mono_coop_sem_timedwait (sem=0x4bf4e88 <worker+72>, flags=MONO_SEM_FLAGS_ALERTABLE, timeout_ms=<optimized out>) at ../../mono/utils/mono-coop-semaphore.h:55
#4  worker_park () at threadpool-worker-default.c:391
#5  worker_thread (unused=<optimized out>) at threadpool-worker-default.c:490
#6  worker_thread (unused=unused@entry=0x0) at threadpool-worker-default.c:457
#7  0x0000000000ea1a55 in start_wrapper_internal (stack_ptr=0x7f57eae9f000, start_info=0x0) at threads.c:1226
#8  start_wrapper (data=0x7f57d40010d0) at threads.c:1299
#9  0x00007f5839684fa3 in start_thread () from /lib/x86_64-linux-gnu/libpthread.so.0
#10 0x00007f58394234cf in clone () from /lib/x86_64-linux-gnu/libc.so.6

Thread 13 (Thread 0x7f57eb09f700 (LWP 10452)):
#0  0x00007f583968da82 in do_futex_wait () from /lib/x86_64-linux-gnu/libpthread.so.0
#1  0x00007f583968db93 in __new_sem_wait_slow () from /lib/x86_64-linux-gnu/libpthread.so.0
#2  0x0000000000e91154 in mono_os_sem_timedwait (flags=MONO_SEM_FLAGS_ALERTABLE, timeout_ms=30797, sem=0x4bf4e88 <worker+72>) at ../../mono/utils/mono-os-semaphore.h:252
#3  mono_coop_sem_timedwait (sem=0x4bf4e88 <worker+72>, flags=MONO_SEM_FLAGS_ALERTABLE, timeout_ms=<optimized out>) at ../../mono/utils/mono-coop-semaphore.h:55
#4  worker_park () at threadpool-worker-default.c:391
#5  worker_thread (unused=<optimized out>) at threadpool-worker-default.c:490
#6  worker_thread (unused=unused@entry=0x0) at threadpool-worker-default.c:457
#7  0x0000000000ea1a55 in start_wrapper_internal (stack_ptr=0x7f57eb0a0000, start_info=0x0) at threads.c:1226
#8  start_wrapper (data=0xed824f0) at threads.c:1299
#9  0x00007f5839684fa3 in start_thread () from /lib/x86_64-linux-gnu/libpthread.so.0
#10 0x00007f58394234cf in clone () from /lib/x86_64-linux-gnu/libc.so.6

Thread 12 (Thread 0x7f5806c56700 (LWP 10278)):
#0  0x00007f583968d896 in do_futex_wait.constprop () from /lib/x86_64-linux-gnu/libpthread.so.0
#1  0x00007f583968d988 in __new_sem_wait_slow.constprop.0 () from /lib/x86_64-linux-gnu/libpthread.so.0
#2  0x0000000002965b2d in SemaphorePosix::wait() [clone .part.0] ()
#3  0x0000000006e8e630 in ?? ()
#4  0x000000000274ed13 in EditorResourcePreview::_thread() ()
#5  0x0000000002a542c5 in ThreadPosix::thread_callback(void*) ()
#6  0x0000000000000000 in ?? ()

Thread 11 (Thread 0x7f5807531700 (LWP 10274)):
#0  0x00007f583968ebf0 in nanosleep () from /lib/x86_64-linux-gnu/libpthread.so.0
#1  0x0000000000f74c0b in OS_Unix::delay_usec(unsigned int) const [clone .constprop.12944] ()
#2  0x0000000000000000 in ?? ()

Thread 10 (Thread 0x7f5807d37700 (LWP 10273)):
#0  0x00007f583968ebf0 in nanosleep () from /lib/x86_64-linux-gnu/libpthread.so.0
#1  0x0000000000f74c0b in OS_Unix::delay_usec(unsigned int) const [clone .constprop.12944] ()
#2  0x0000000000000000 in ?? ()

Thread 9 (Thread 0x7f5807d7e700 (LWP 10272)):
#0  0x00007f583968ebf0 in nanosleep () from /lib/x86_64-linux-gnu/libpthread.so.0
#1  0x0000000000f74c0b in OS_Unix::delay_usec(unsigned int) const [clone .constprop.12944] ()
#2  0x0000000000000000 in ?? ()

Thread 8 (Thread 0x7f5828ffe700 (LWP 10271)):
#0  0x00007f583968d896 in do_futex_wait.constprop () from /lib/x86_64-linux-gnu/libpthread.so.0
#1  0x00007f583968d988 in __new_sem_wait_slow.constprop.0 () from /lib/x86_64-linux-gnu/libpthread.so.0
#2  0x0000000000dcdd78 in mono_os_sem_wait (flags=MONO_SEM_FLAGS_ALERTABLE, sem=0x4bed4e0 <finalizer_sem>) at ../../mono/utils/mono-os-semaphore.h:203
#3  mono_coop_sem_wait (flags=MONO_SEM_FLAGS_ALERTABLE, sem=0x4bed4e0 <finalizer_sem>) at ../../mono/utils/mono-coop-semaphore.h:41
#4  finalizer_thread (unused=unused@entry=0x0) at gc.c:967
#5  0x0000000000ea1a55 in start_wrapper_internal (stack_ptr=0x7f5828fff000, start_info=0x0) at threads.c:1226
#6  start_wrapper (data=0x6721d00) at threads.c:1299
#7  0x00007f5839684fa3 in start_thread () from /lib/x86_64-linux-gnu/libpthread.so.0
#8  0x00007f58394234cf in clone () from /lib/x86_64-linux-gnu/libc.so.6

Thread 7 (Thread 0x7f582a7ff700 (LWP 10270)):
#0  0x00007f583968b00c in pthread_cond_wait@@GLIBC_2.3.2 () from /lib/x86_64-linux-gnu/libpthread.so.0
#1  0x0000000000f0c556 in mono_os_cond_wait (mutex=0x4bffd60 <lock>, cond=0x4bffd20 <work_cond>) at ../../mono/utils/mono-os-mutex.h:177
#2  get_work (job=<synthetic pointer>, do_idle=<synthetic pointer>, work_context=<synthetic pointer>, worker_index=0) at sgen-thread-pool.c:165
#3  thread_func (data=<optimized out>) at sgen-thread-pool.c:196
#4  0x00007f5839684fa3 in start_thread () from /lib/x86_64-linux-gnu/libpthread.so.0
#5  0x00007f58394234cf in clone () from /lib/x86_64-linux-gnu/libc.so.6

Thread 6 (Thread 0x7f5830116700 (LWP 10269)):
#0  0x00007f58393f0720 in nanosleep () from /lib/x86_64-linux-gnu/libc.so.6
#1  0x00007f583941b874 in usleep () from /lib/x86_64-linux-gnu/libc.so.6
#2  0x00000000031f5f02 in JoypadLinux::monitor_joypads() ()
#3  0x000000770000007c in ?? ()
#4  0x0000000005f8b640 in ?? ()
#5  0x0000000000000018 in ?? ()
#6  0x00007f580c000b50 in ?? ()
#7  0x706e692f7665642f in ?? ()
#8  0x746e6576652f7475 in ?? ()
#9  0xffffffffff003133 in ?? ()
#10 0x00007f5830116700 in ?? ()
#11 0x0000000005cecd20 in ?? ()
#12 0x00007f58393ae56a in malloc () from /lib/x86_64-linux-gnu/libc.so.6
#13 0x00007ffeb841fbde in ?? ()
#14 0x00007ffeb841fbdf in ?? ()
#15 0x00007f5830116700 in ?? ()
#16 0x0000000005cecd20 in ?? ()
#17 0x0000000002a542c5 in ThreadPosix::thread_callback(void*) ()
#18 0x0000000000000000 in ?? ()

Thread 5 (Thread 0x7f5830157700 (LWP 10268)):
#0  0x00007f583968ebf0 in nanosleep () from /lib/x86_64-linux-gnu/libpthread.so.0
#1  0x0000000000f74c0b in OS_Unix::delay_usec(unsigned int) const [clone .constprop.12944] ()
#2  0x0000000000000000 in ?? ()

Thread 4 (Thread 0x7f5830599700 (LWP 10267)):
#0  0x00007f583968d896 in do_futex_wait.constprop () from /lib/x86_64-linux-gnu/libpthread.so.0
#1  0x00007f583968d988 in __new_sem_wait_slow.constprop.0 () from /lib/x86_64-linux-gnu/libpthread.so.0
#2  0x0000000002965b2d in SemaphorePosix::wait() [clone .part.0] ()
#3  0x00007f583059a020 in ?? ()
#4  0x00000000015f5a83 in VisualServerScene::_gi_probe_bake_thread() ()
#5  0x0000000000000008 in ?? ()
#6  0x0000000005c53710 in ?? ()
#7  0x0000000000000000 in ?? ()

Thread 3 (Thread 0x7f583119f700 (LWP 10266)):
#0  0x00007f583968b00c in pthread_cond_wait@@GLIBC_2.3.2 () from /lib/x86_64-linux-gnu/libpthread.so.0
#1  0x00007f5831b55e83 in ?? () from /usr/lib/x86_64-linux-gnu/dri/i965_dri.so
#2  0x00007f5831b55bd7 in ?? () from /usr/lib/x86_64-linux-gnu/dri/i965_dri.so
#3  0x00007f5839684fa3 in start_thread () from /lib/x86_64-linux-gnu/libpthread.so.0
#4  0x00007f58394234cf in clone () from /lib/x86_64-linux-gnu/libc.so.6

Thread 2 (Thread 0x7f5832a46700 (LWP 10261)):
#0  0x00007f583968d896 in do_futex_wait.constprop () from /lib/x86_64-linux-gnu/libpthread.so.0
#1  0x00007f583968d988 in __new_sem_wait_slow.constprop.0 () from /lib/x86_64-linux-gnu/libpthread.so.0
#2  0x0000000002965b2d in SemaphorePosix::wait() [clone .part.0] ()
#3  0x00000000057d5fc0 in ?? ()
#4  0x00000000010ce9c4 in _IP_ResolverPrivate::_thread_function(void*) ()
#5  0x00007ffeb841ff60 in ?? ()
#6  0x00007f58393ae56a in malloc () from /lib/x86_64-linux-gnu/libc.so.6
#7  0x00007ffeb841fe9e in ?? ()
#8  0x00007ffeb841fe9f in ?? ()
#9  0x00007f5832a46700 in ?? ()
#10 0x00007ffeb841ff60 in ?? ()
#11 0x0000000002a542c5 in ThreadPosix::thread_callback(void*) ()
#12 0x0000000000000000 in ?? ()

Thread 1 (Thread 0x7f58370fc880 (LWP 10256)):
#0  0x00007f583968f0ca in waitpid () from /lib/x86_64-linux-gnu/libpthread.so.0
#1  0x0000000000d10d71 in dump_native_stacktrace (mctx=mctx@entry=0x7ffeb841f270, signal=0x338e627 "SIGSEGV") at mini-posix.c:1089
#2  0x0000000000d10efc in mono_dump_native_crash_info (signal=signal@entry=0x338e627 "SIGSEGV", mctx=mctx@entry=0x7ffeb841f270, info=info@entry=0x0) at mini-posix.c:1135
#3  0x0000000000d039a8 in mono_handle_native_crash (signal=signal@entry=0x338e627 "SIGSEGV", mctx=mctx@entry=0x7ffeb841f270, info=info@entry=0x0) at mini-exceptions.c:3423
#4  0x0000000000c7c211 in altstack_handle_and_restore (ctx=0x7ffeb841f270, obj=0x0, flags=<optimized out>) at exceptions-amd64.c:875
#5  0x00000000030175a0 in ?? ()
#6  0x0000000000000000 in ?? ()
[Inferior 1 (process 10256) detached]

=================================================================
	Basic Fault Address Reporting
=================================================================
Memory around native instruction pointer (0x30175a0):0x3017590  48 8b 7f 10 e9 97 3c d8 fd 90 66 0f 1f 44 00 00  H.....<...f..D..
0x30175a0  48 8b 7f 10 e9 77 4c d9 fd 90 66 0f 1f 44 00 00  H....wL...f..D..
0x30175b0  53 48 89 fb e8 e7 ff ff ff 48 89 c7 e8 af a0 e1  SH.......H......
0x30175c0  fd 48 89 da 5b 89 c0 c3 0f 1f 84 00 00 00 00 00  .H..[...........

=================================================================
	Managed Stacktrace:
=================================================================
	  at <unknown> <0xffffffff>
	  at Godot.NativeCalls:godot_icall_Node2D_Ctor <0x00074>
	  at Godot.Node2D:.ctor <0x0003b>
	  at Entity:.ctor <0x001a3>
	  at RetroMover:.ctor <0x00423>
	  at Player:.ctor <0x00277>
	  at System.Object:runtime_invoke_void__this__ <0x00071>
=================================================================
```

**Steps to reproduce:**
I'm not sure about the exact cause of the crash, but using a sorted list with a custom comparer makes the editor crash consistently after modifying the code and building the project solution.

**Minimal reproduction project:**
[MonoCrashMRP.zip](https://github.com/godotengine/godot/files/4513021/MonoCrashMRP.zip)

**Comment by aaronfranke** [comment]
I've added ``` around your output, but consider using a service such as Pastebin instead in the future.

**Comment by MightyPrinny** [comment]
I've added the MRP, maybe @neikeq can figure it out now.

**Comment by neikeq** [comment]
I think this no longer happens after #38141. Let's wait for the next an updated build of 3.2.2.

**Comment by akien-mga** [comment]
Here's 3.2.2 beta 2 which includes #38141, could you confirm that it fixes it? https://downloads.tuxfamily.org/godotengine/3.2.2/beta2/mono/

**Comment by MightyPrinny** [comment]
The mrp doesn't make the editor crash in beta 2.

**Comment by akien-mga** [comment]
Great, closing as fixed by #38141.
